### PR TITLE
Add context-splitting for the string "none"

### LINKED ISSUE #22094: Context-splitting needed for the string "none"
The strings "None" and "none" are used in many places.
Unfortunately, this string is impossible to translate correctly in many languages.
The reason is that depending on what noun is described, this word may need to be presented in various forms.
For reference, [see the original ticket on Trac](https://core.trac.wordpress.org/ticket/50087).

## How to fix it
replace `__( 'None' )` occurrences with `_x( 'None', 'Some context )`

I'll submit a PR in few mins.

### PR DESCRIPTION
**Pull request by audrasjb**
## Description
The strings "None" and "none" are used in many places.
Unfortunately, this string is impossible to translate correctly in many languages.
The reason is that depending on what noun is described, this word may need to be presented in various forms.
For reference, see the original ticket on Trac.

This changeset replaces `__( 'None' )` occurrences with `_x( 'None', 'Some context )`

Fixes #22094

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by audrasjb** [comment]
Fixes #22094

**Comment by audrasjb** [comment]
I fixed one of the 3 linting errors, but I can't see what I need to do to fix the two last ones. Could someone explain me the issue please? thanks 🙂

**Comment by audrasjb** [comment]
Ok checks are green now 🍏 🙂

**Comment by youknowriad** [comment]
Seems good but I don't have a good i18n knowledge to judge the context splitting  here. ccc @swissspidy

File: packages/block-library/src/button/edit.native.js [hunk 1]
```diff
@@ -343,7 +343,7 @@ class ButtonEdit extends Component {
 					icon={ ! isCompatibleWithSettings && LinkRelIcon }
 					label={ __( 'Link Rel' ) }
 					value={ rel || '' }
-					valuePlaceholder={ __( 'None' ) }
+					valuePlaceholder={ _x( 'None', 'Rel attribute value' ) }
```
Comment (swissspidy) [inline packages/block-library/src/button/edit.native.js:346]:
```suggestion
					valuePlaceholder={ _x( 'None', 'Link rel attribute value placeholder' ) }
```

**Comment by swissspidy** [comment]
One suggestion and 1 conflict that needs to be resolved, otherwise LGTM

File: packages/block-library/src/audio/edit.js [hunk 2]
```diff
@@ -183,7 +183,10 @@ class AudioEdit extends Component {
 								{ value: '', label: __( 'Browser default' ) },
 								{ value: 'auto', label: __( 'Auto' ) },
 								{ value: 'metadata', label: __( 'Metadata' ) },
-								{ value: 'none', label: __( 'None' ) },
+								{
+									value: 'none',
+									label: _x( 'None', 'Preload value' ),
```
Comment (mcsf) [inline packages/block-library/src/audio/edit.js:188]: Perhaps:

```suggestion
									label: _x( 'None', '"Preload" value' ),
```

would make it more explicit.

I was also going to suggest appending "for Audio block" but I see that Video is meant to reuse this context.

Comment (mcsf) [reply packages/block-library/src/audio/edit.js:188]: Speaking of which, that `__( 'Preload' )` just a few lines above seems like a quick win to add to this PR. 😉 

`_x( 'Preload', 'noun; Audio block parameter' )`

**Review by mcsf — APPROVED** [review]

**Comment by aristath** [comment]
I merged trunk and resolved conflicts in https://github.com/WordPress/gutenberg/compare/trunk...fix/contextualize-none-strings
@audrasjb you can merge that branch in your fork if you want to update it (https://github.com/audrasjb/gutenberg/compare/fix/contextualize-none-strings...WordPress:fix/contextualize-none-strings), this one would be nice to land :+1: :ship:

**Comment by ramonjd** [comment]
@audrasjb I'm happy to carry this one forward if you don't have time. Agree with other folks that it would be good to land to make life for our translators a little easier 😄

**Comment by audrasjb** [comment]
Thank you for the proposal @ramonjd. Very appreciated cause I'm currently on holidays 😅

**Comment by ramonjd** [comment]
> Thank you for the proposal @ramonjd. Very appreciated cause I'm currently on holidays 😅

Nice! Hope you have a great time. Wish I were on holidays too 🤣 

I'll spin up a new PR soon. Thanks for your work on this.

**Comment by ramonjd** [comment]
Updated PR https://github.com/WordPress/gutenberg/pull/34341

Closing this one. 


Thank you!

**Pull request closed without merging**